src/styles/tailwind.css

```diff
@@ -1,3 +1,110 @@
+@import "tailwindcss/base";
+@import "tailwindcss/components";
+@import "tailwindcss/utilities";
+
+@layer base {
+  body {
+    /* preDefine CSS variables in Obsidian.(https://docs.obsidian.md/Reference/CSS+variables/Foundations/Colors) */
+    /*Primary background*/
+    --ob-bg-primary: var(--background-primary);
```

Comment:
why do we need to redefine obsidian CSS with `--ob-bg` prefix?

Reply:
I think that  redefine obsidian CSS will more clarify. I want to provide out-of-the-box functionality, when we add some shadcn/ui component.

Reply:
Do you have any specific features in mind that would require us to break apart from the default css variables? This is an additional layer that we have to maintain. If we want to apply a new obsidian css variable that is not included in this PR, the process of adding it would require us to:
1. Make a copy of the css variable 
2. Create a new tailwind token with a different semantic value 

The more undocumented overhead we add, the more difficult it is for new contributors to make changes. 

If you feel strongly that creating our own semantic token is necessary, and it can simplify the component library adoption, I'm supportive of that. However, we should at least consider not adding another layer of complexity and use the obsidian css variable directly.

Also worth noting that any external CSS variable that we introduce will not be compatible with the obsidian theme, which can be a dealbreaker for some users.

Reply:
I am merely listing all the obsidian css variables rather than defining a new set of variables.

> If we want to apply a new obsidian css variable

I have listed most of the color css variables.

> The more undocumented overhead we add, the more difficult it is for new contributors to make changes.

When a new contributor joins, he/her doesn't need to do anything extra work, and he/her doesn't need to care about these. 

He can use existing components or download new ones from the shadcn/ui, and then he doesn't need to do anything, the component will automatically change with the theme.